-1

Tested at: a1233a12523484d4b77a90259be269dfa85f4b27

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16664f/1307/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/DQMOffline/RecoB/src/JetTagPlotter.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/DQMOffline/RecoB/src/TrackCountingTagPlotter.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/DQMOffline/RecoB/src/BTagDifferentialPlot.cc 
In file included from /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/DQMOffline/RecoB/interface/MVAJetTagPlotter.h:10:0,
                 from /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/DQMOffline/RecoB/src/TagInfoPlotterFactory.cc:11:
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/RecoBTau/JetTagComputer/interface/GenericMVAJetTagComputer.h:20:70: error: 'BTauGenericMVAJetTagComputerRcd' was not declared in this scope
     edm::ESGetToken<PhysicsTools::Calibration::MVAComputerContainer, BTauGenericMVAJetTagComputerRcd> calib_;
                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-03-1100/src/RecoBTau/JetTagComputer/interface/GenericMVAJetTagComputer.h:20:70: note: suggested alternative: 'GenericMVAJetTagComputer'
     edm::ESGetToken<PhysicsTools::Calibration::MVAComputerContainer, BTauGenericMVAJetTagComputerRcd> calib_;
                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</pre>

--- a/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h
+++ b/RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h
@@ -9,6 +9,7 @@
 #include "FWCore/Framework/interface/ESConsumesCollector.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "CommonTools/MVAUtils/interface/TMVAEvaluator.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
 #include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
 
 class CombinedMVAV2JetTagComputer : public JetTagComputer {

--- a/RecoBTag/Combined/src/CombinedMVAV2JetTagComputer.cc
+++ b/RecoBTag/Combined/src/CombinedMVAV2JetTagComputer.cc
@@ -12,9 +12,6 @@
 #include "DataFormats/JetReco/interface/Jet.h"
 #include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
 #include "RecoBTag/Combined/interface/CombinedMVAV2JetTagComputer.h"
-#include "CondFormats/DataRecord/interface/BTauGenericMVAJetTagComputerRcd.h"
-#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
-#include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
 
 using namespace reco;
 

--- a/RecoBTag/SoftLepton/interface/MuonTagger.h
+++ b/RecoBTag/SoftLepton/interface/MuonTagger.h
@@ -8,6 +8,7 @@
 #include "FWCore/Framework/interface/ESConsumesCollector.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "CommonTools/MVAUtils/interface/TMVAEvaluator.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
 #include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
 #include "RecoBTag/SoftLepton/interface/LeptonSelector.h"
 #include <memory>

--- a/RecoBTag/SoftLepton/src/MuonTagger.cc
+++ b/RecoBTag/SoftLepton/src/MuonTagger.cc
@@ -5,8 +5,6 @@
 #include <limits>
 #include <random>
 
-#include "CondFormats/DataRecord/interface/BTauGenericMVAJetTagComputerRcd.h"
-#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
 #include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
 #include "DataFormats/BTauReco/interface/SoftLeptonTagInfo.h"
 #include "DataFormats/BTauReco/interface/CandSoftLeptonTagInfo.h"

--- a/RecoBTau/JetTagComputer/interface/GenericMVAJetTagComputer.h
+++ b/RecoBTau/JetTagComputer/interface/GenericMVAJetTagComputer.h
@@ -5,6 +5,7 @@
 
 #include "FWCore/Framework/interface/ESConsumesCollector.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/DataRecord/interface/BTauGenericMVAJetTagComputerRcd.h"
 #include "DataFormats/BTauReco/interface/BaseTagInfo.h"
 #include "DataFormats/BTauReco/interface/TaggingVariable.h"
 #include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
@@ -17,7 +18,7 @@ class GenericMVAJetTagComputer : public JetTagComputer {
 public:
   struct Tokens {
     Tokens(const edm::ParameterSet &parameters, edm::ESConsumesCollector &&cc);
-    edm::ESGetToken<PhysicsTools::Calibration::MVAComputerContainer, JetTagComputerRecord> calib_;
+    edm::ESGetToken<PhysicsTools::Calibration::MVAComputerContainer, BTauGenericMVAJetTagComputerRcd> calib_;
   };
 
   GenericMVAJetTagComputer(const edm::ParameterSet &parameters, Tokens tokens);

--- a/RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h
+++ b/RecoBTau/JetTagComputer/interface/JetTagComputerESProducer.h
@@ -11,7 +11,7 @@
 #include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
 
 /**
- * The idea here is to provide to implementations for
+ * The idea here is to provide two implementations for
  * JetTagConmputerESProducer: one for those ConcreteJetTagComputers
  * that consume ES products and thus need the ESGetTokens, and one for
  * those that do not. All ConcreteJetTagComputers are required to have
@@ -26,7 +26,7 @@
  * Those that do not need ESGetTokens should define the nested type as
  * void, and in this case no further modifications are needed.
  */
-namespace jetTagComputerESProducerImpl {
+namespace jet_tag_computer_esproducer_impl {
   template <typename ConcreteJetTagComputer, bool>
   class JetTagComputerESProducer : public edm::ESProducer {
   private:
@@ -72,10 +72,10 @@ namespace jetTagComputerESProducerImpl {
   private:
     const edm::ParameterSet m_pset;
   };
-}  // namespace jetTagComputerESProducerImpl
+}  // namespace jet_tag_computer_esproducer_impl
 
 template <typename T>
 using JetTagComputerESProducer =
-    jetTagComputerESProducerImpl::JetTagComputerESProducer<T, std::is_same_v<typename T::Tokens, void>>;
+    jet_tag_computer_esproducer_impl::JetTagComputerESProducer<T, std::is_same_v<typename T::Tokens, void>>;
 
 #endif  // RecoBTau_JetTagComputerESProducer_h

--- a/RecoBTau/JetTagComputer/src/GenericMVAJetTagComputer.cc
+++ b/RecoBTau/JetTagComputer/src/GenericMVAJetTagComputer.cc
@@ -6,7 +6,6 @@
 #include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "CondFormats/PhysicsToolsObjects/interface/MVAComputer.h"
-#include "CondFormats/DataRecord/interface/BTauGenericMVAJetTagComputerRcd.h"
 #include "DataFormats/Common/interface/RefToBase.h"
 #include "DataFormats/JetReco/interface/Jet.h"
 #include "DataFormats/BTauReco/interface/TaggingVariable.h"

